src/coreComponents/codingUtilities/Utilities.hpp

```diff
@@ -414,6 +414,51 @@ struct BitFlags
 
 };
 
+template< typename VECTOR_TYPE, typename SCALAR_TYPE >
+VECTOR_TYPE addTwoVecs( VECTOR_TYPE const & vec1,
+                        VECTOR_TYPE const & vec2,
+                        SCALAR_TYPE const sign )
+{
+  GEOS_ASSERT( vec1.size() == vec2.size() );
+  const localIndex N = vec1.size();
+  VECTOR_TYPE result( N );
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                          [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result[i] = vec1[i] + sign * vec2[i];
+  } );
+  return result;
+}
+
+template< typename VECTOR_TYPE, typename SCALAR_TYPE >
+VECTOR_TYPE scalarMultiplyAVec( VECTOR_TYPE const & vec,
+                                SCALAR_TYPE const scalarMult )
+{
+  const localIndex N = vec.size();
+  VECTOR_TYPE result( N );
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                          [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result[i] = scalarMult * vec[i];
+  } );
+  return result;
+}
+
+template< typename VECTOR_TYPE >
+real64 dotTwoVecs( VECTOR_TYPE const & vec1,
+                   VECTOR_TYPE const & vec2 )
+{
+  GEOS_ASSERT( vec1.size() == vec2.size());
+  real64 result = 0;
+  const localIndex N = vec1.size();
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                          [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result += vec1[i] * vec2[i];
```

Comment:
This will produce the wrong result in parallel, you need `RAJA::ReduceSum`

Reply:
yes sure, sorry